### ISSUE TITLE
Need better error message if Internet connection is lost during project import

### ISSUE THREAD

**Issue by cckozie** [unfoldingWord/translationCore#1373]
0.6.17
Windows error message:
![image](https://cloud.githubusercontent.com/assets/8171759/25588524/3f8e477e-2e6e-11e7-87e0-c4440ef98cf7.png)

**Comment by benjore** [comment]
#1373 #1257 #1246 #1393 #1386 

- [ ] Figure out way to handle internet error messages

**Comment by cckozie** [comment]
Error message should be:
"Unable to connect to the server. Please check your Internet connection.".

**Comment by cdwhitfield73** [comment]
I think I've covered the lost internet issue.  Also set up a generic default message so the user will never get "code speak" in the dialog.  There may be more scenarios that we want to add specific messages for down the road.

**Comment by cckozie** [comment]
I cut off the Internet during the download. I received no error message. Instead I eventually got this screen indicating both that I needed to load a project and my project would be loaded after choosing a tool.
![image](https://cloud.githubusercontent.com/assets/8171759/25764951/0076e704-31b0-11e7-908e-59b866c7eae1.png)

**Comment by cdwhitfield73** [comment]
Major changes to importing functions.  Many issues addressed.  Hopefully got them all.

**Comment by cckozie** [comment]
@cdwhitfield73 In my testing of this issue, about half of the time I get the no internet connection message and half the time I get the unknown problem message. Any idea on why that is?

**Comment by cdwhitfield73** [comment]
So, how the error handling function works is that it starts with the "Unknown error" message as the default.  Then it does a check on the actual error it received for several different "known" messages and then converts them into "user friendly terminology".  Whenever I lost internet connection on my testing, it always gave a similar error, so that is the one I am looking for.  Apparently, some of the time it is sending back a slightly different error that my checks aren't recognizing.  The only way to test this is for me to add a console.log so it shows the real error in the console and then when you get the unknown error message, you send me a screenshot of the console so I can see what you are getting.

**Comment by cckozie** [comment]
@cdwhitfield73 I've also seen the issue (only once so far) where I get no message, the spinner just kept going for several minutes until I finally stopped it.  Any idea on why that might happen? Unfortunately I deleted my screenshot of the console.
I did capture this when I got the unknown error message. Is that helpful at all? I don't think this is worth taking a lot of time on at this point, but if it's something fairly obvious it would good to fix it now.
![image](https://cloud.githubusercontent.com/assets/8171759/25973287/0e497046-3669-11e7-99c4-e8566a3f568e.png)

**Comment by cdwhitfield73** [comment]
@cckozie that helps.  I am catching the "fatal: unable to access" but not the "fatal: remote end hung up" or the ERR_DISCONNECTED.  I can add those to the list and it should help.  Not sure about the one where it just spun forever.

**Comment by cdwhitfield73** [comment]
submitted a PR that checks for the additional errors.  Retest with next build (24)

**Comment by cckozie** [comment]
@cdwhitfield73 
I've seen a couple more interesting scenarios. 
1. There is a difference between Windows and Mac in some error cases. See second comment in #1556.
2. With that issue pending, when I loose the Internet connection when using the URL to download on my Mac, the wheel spins for several minutes (9+) with no message. Eventually I do get the Unable to connect to the server... message. This is what I see in the console:
![image](https://cloud.githubusercontent.com/assets/8171759/26079374/34cc28c2-3988-11e7-9fe3-2ecefe65b5ee.png)
All of my previous testing was on Windows.

**Comment by cckozie** [comment]
If the Internet connection is restored during the time that the wheel is spinning, sometimes the download completes successfully and sometimes the Unable to connect to the server... message is displayed.

**Comment by cckozie** [comment]
I'm going to pass this and write up a new issue for 2. in the comment above.
